homeassistant/components/garmin_connect/__init__.py

```diff
@@ -90,6 +90,16 @@ def __init__(self, hass, client):
         self.client = client
         self.data = None
 
+    async def _get_combined_alarms_of_all_devices(self):
```

Comment:
Docstring is missing

Reply:
for a private method?

Reply:
done

Reply:
We should make this a sync function and schedule the whole function on the executor instead. That will avoid the repetitive context switches, async to sync.

Reply:
the problem with that, if I understand you correctly, is that Garmin is pulled every 10 minutes
so if I have several alarms sequenced in that period they will never show up

Reply:
I'm not following.